src/test/java/org/folio/rest/support/kafka/FakeKafkaConsumer.java

```diff
@@ -216,14 +244,18 @@ private static String instanceAndIdKey(KafkaConsumerRecord<String, JsonObject> m
     return instanceAndIdKey(message.key(), id);
   }
 
+  private void setConsumer(KafkaConsumer<String, JsonObject> consumer) {
+    this.consumer = consumer;
+  }
```

Comment:
What benefit does this provide over interacting with the field directly?

Reply:
The original code for `public FakeKafkaConsumer consume(Vertx vertx)` creates the  consumer within the method.
There is no access to the `consumer` after that function call and there is the potential for the consumer being a weak reference.

Make the consumer a strong reference.
This is to help avoid the possibility of the `consumer` being closed before everything is done with it.
This appears to help solve several of the connection closed warnings/errors.

This also allows for the following:
```
  public void unsubscribe() {
    consumer.unsubscribe();
  }
```

The consumer is on the class and may now be explicitly unsubscribed.

The `TestBase` then can do this:
```
  @AfterClass
  public static void afterAll() {
    kafkaConsumer.unsubscribe();
  }
```

The `consumer` should be unsubscribed after each class if we are always subscribing to a new one before each class.
This appears to help solve several of the connection closed warnings/errors.

Reply:
> The consumer should be unsubscribed after each class if we are always subscribing to a new one before each class.
This appears to help solve several of the connection closed warnings/errors.

Hmm, that should only really be necessary when a different Kafka instance is used. That suggests that we don't have alignment between Kafka deployments and tests.

Reply:
Maybe a comment could be added to explain the use of the strong reference and explicit disconnection (though I do prefer could being explicit, so I'm good with this change).

Maybe in the future, this class becomes instantiated within the tests and not shared.

Reply:
Comments added.